src/main/java/com/welovecoding/netbeans/plugin/editorconfig/editor/api/lexer/ECTokenId.java

```diff
@@ -0,0 +1,139 @@
+  EOF(null, Category.WHITESPACE, ECTokenId.TOKEN_EOF);
+
+  private enum Category {
+
+    WHITESPACE("whitespace"), // NOI18N
+    KEYWORD("keyword"), // NOI18N
+    SECTION("section"), // NOI18N
+    SECTION_DELIMITER("section_delimiter"), // NOI18N
+    COMMENT("comment"), // NOI18N
+    STRING("string"), // NOI18N
+    EQUALS("equals"); // NOI18N
+    private final String name;
+
+    Category(String name) {
+      this.name = name;
+    }
+
+    public String getName() {
+      return name;
+    }
+  }
+
+  public static final int TOKEN_WS = 1;
+  public static final int TOKEN_PROPERTY_KEY_ROOT = 2;
+  public static final int TOKEN_PROPERTY_KEY_CHARSET = 3;
+  public static final int TOKEN_PROPERTY_KEY_END_OF_LINE = 4;
+  public static final int TOKEN_PROPERTY_KEY_INDENT_SIZE = 5;
+  public static final int TOKEN_PROPERTY_KEY_INDENT_STYLE = 6;
+  public static final int TOKEN_PROPERTY_KEY_INSERT_FINAL_NEWLINE = 7;
+  public static final int TOKEN_PROPERTY_KEY_MAX_LINE_LENGTH = 8;
+  public static final int TOKEN_PROPERTY_KEY_TAB_WIDTH = 9;
+  public static final int TOKEN_PROPERTY_KEY_TRIM_TRAILING_WHITESPACE = 10;
+  public static final int TOKEN_PROPERTY_KEY_UNKNOWN = 11;
+  public static final int TOKEN_SECTION_LDELIMITER = 12;
+  public static final int TOKEN_SECTION_RDELIMITER = 13;
+  public static final int TOKEN_ASSIGNMENT = 14;
+  public static final int TOKEN_COMMENT = 15;
+  public static final int TOKEN_TEXT = 16;
+  public static final int TOKEN_SECTION_NAME = 17;
+  public static final int TOKEN_PROPERTY_VALUE = 18;
+  public static final int TOKEN_EOF = -1;
+
+  public static final String EDITORCONFIG_MIME_TYPE = "text/x-editorconfig"; // NOI18N
+  private final String fixedText;
+  private final Category primaryCategory;
+  private final int id;
+  private static final Map<Integer, ECTokenId> TOKENS = Collections.synchronizedMap(new HashMap<Integer, ECTokenId>());
+
+  static {
+    for (ECTokenId value : values()) {
+      TOKENS.put(value.getId(), value);
+    }
+  }
+
+  ECTokenId(String fixedText, Category primaryCategory, int id) {
+    this.fixedText = fixedText;
+    this.primaryCategory = primaryCategory;
+    this.id = id;
+  }
+
+  @Override
+  public String primaryCategory() {
+    return primaryCategory.getName();
+  }
+
+  public String getFixedText() {
+    return fixedText;
+  }
+
+  public int getId() {
+    return id;
+  }
+
+  public static ECTokenId toEnum(int tokenType) {
+    return TOKENS.get(tokenType);
+  }
+
+  private static final Language<ECTokenId> EDITORCONFIG_LANGUAGE = new LanguageHierarchy<ECTokenId>() {
+
+    @Override
+    protected Collection<ECTokenId> createTokenIds() {
+      return EnumSet.allOf(ECTokenId.class);
+    }
+
+    @Override
+    protected Lexer<ECTokenId> createLexer(LexerRestartInfo<ECTokenId> info) {
+      return ECLexer.create(info);
+    }
+
+    @Override
+    protected String mimeType() {
+      return EDITORCONFIG_MIME_TYPE;
```

Comment:
:+1: 

May I add a few changes to Settings.java?
- private constructor
- final modifier